spring-core/src/main/java/org/springframework/core/GenericTypeResolver.java

```diff
@@ -180,7 +180,7 @@ else if (genericType instanceof ParameterizedType parameterizedType) {
 								generics[i] = resolvedTypeArgument;
 							}
 							else {
-								generics[i] = ResolvableType.forType(typeArgument);
+								generics[i] = ResolvableType.forType(typeArgument).resolveType();
```

Comment:
as far as I understand the code here, we want to resolve the `T` of a generic. If the `T` is not defined by the contextType, (because ist is no `class Foo<T extends Bar>`) we get to this line.

In my case, the T is bound by the function `public <T extends BaseType> List<T> get()` and means, that `T` is at least a `BaseType`

IMHO the type must be resolved here, so that the following code knows , that the elements in the list are `BaseType` or subtypes. This is later required for jackson, otherwise we have problems mentioned in this [SO](https://stackoverflow.com/questions/34193177/why-does-jackson-polymorphic-serialization-not-work-in-lists) post.